init.lua

```diff
@@ -0,0 +1,576 @@
+vim.opt.ignorecase = true
+vim.opt.smartcase = true
+vim.opt.wildignorecase = true
+
+-- Ripgrep (if available)
+if vim.fn.executable("rg") == 1 then
+  vim.opt.grepformat = "%f:%l:%c:%m"
+  vim.opt.grepprg = "rg --vimgrep -i"
+end
+
+-- Undo history
+vim.opt.undodir = vim.fs.joinpath(vim.fn.stdpath("state"), "undodir")
+vim.opt.undofile = true
+
+-- Fallback colorscheme
+vim.cmd.colorscheme("habamax")
+
+-- Diagnotics
+vim.diagnostic.config({
+  underline = false,
+  severity_sort = true,
+})
+
+---}}}
+
+---{{{ Keymaps
+
+--- This is your leader key.
+---
+--- It will be used for most of your custom keymaps to ensure they don't conflict with anything
+--- that's builtin. It is set to the space key here, but feel free to change it! Many people like
+--- the `,` key as well.
+---
+--- See `:help mapleader` for more information.
+vim.g.mapleader = " "
+
+--- Wrapper function for custom keymaps.
+---
+--- See `:help vim.keymap.set()` for more information.
+local map = function(modes, keys, action, desc)
+  vim.keymap.set(modes, keys, action, { silent = true, desc = desc })
+end
+
+-- File explorer
+map("n", "<Leader>e", vim.cmd.Ex, "Opens netrw.")
+
+-- These two override (in my opinion) bad defaults.
+map("n", "x", "\"_x", "Deletes a single character without yanking it.")
+map("n", "Y", "y$", "Makes `Y` behave more like `D` and `C`.")
+
+-- Yanking and pasting using the system clipboard.
+--
+-- If you're using WSL make sure your Windows clipboard is actually accessible from within WSL.
+-- See `:help clipboard-wsl` for more information.
+--
+-- If you're on Linux make sure to install either `xclip` or `wl-clipboard` (depending on whether
+-- you use X11 or Wayland).
+
+map({ "n", "v" }, "<Leader>y", "\"+y", "Yanks to the system clipboard.")
+map({ "n", "v" }, "<Leader>p", "\"+p", "Pastes from the system clipboard.")
+map({ "n", "v" }, "<Leader>P", "\"+P",
+  "Pastes from the system clipboard without yanking the deleted text.")
+
+-- Stay in visual mode when indenting / dedenting text
+map("x", ">", ">gv", "Indents text in visual mode without losing visual selection.")
+map("x", "<", "<gv", "Dedents text in visual mode without losing visual selection.")
+
+-- Move lines around
+map("n", "<A-j>", ":m .+1<CR>==", "Moves the current line down by 1.")
+map("n", "<A-k>", ":m .-2<CR>==", "Moves the current line up by 1.")
+map("x", "<A-j>", ":m '>+1<CR>=gv", "Moves the current line down by 1.")
+map("x", "<A-k>", ":m '<-2<CR>=gv", "Moves the current line up by 1.")
+
+-- Navigate various lists
+map("n", "[q", ":cprev<CR>", "Jumps to the previous entry in the quickfix list.")
+map("n", "]q", ":cnext<CR>", "Jumps to the next entry in the quickfix list.")
+map("n", "[Q", ":cfirst<CR>", "Jumps to the first entry in the quickfix list.")
+map("n", "]Q", ":clast<CR>", "Jumps to the last entry in the quickfix list.")
+
+map("n", "[b", ":bprev<CR>", "Jumps to the previous buffer.")
+map("n", "]b", ":bnext<CR>", "Jumps to the next buffer.")
```

Comment:
IIRC <kbd>Tab</kbd> and <kbd>Shift + Tab</kbd> in normal mode are not bound to anything, so we could put bprev and bnext on them? :)

Reply:
`<Tab>` and `<C-i>` are the same keycode as far as your terminal is concerned. As `<C-i>` already has functionality (next jump in the jumplist) it would override default behavior, so I'd rather not.

Reply:
Woops :disappointed: